src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,51 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.BulldozerProducer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.ExcavatorProducer;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import core.mate.academy.model.TruckProducer;
+import java.util.ArrayList;
+import java.util.List;
+
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+public class MachineServiceImpl implements MachineService {
```

Comment:
lets parametrize MachineService class

Reply:
Did I understand you correctly, that we need to add <T extends Machine> after MachineServiceImpl ? I did so, but T is never used inside the class and if we do use it, we get a mistake that overriden methods have the same type erasure as methods in MachineService interface